airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -115,6 +115,9 @@ def should_retry(self, response: requests.Response) -> bool:
             setattr(self, "raise_on_http_errors", False)
             self.logger.warning(known_errors.get(status))
             return False
+        elif status == 104:
```

Comment:
In which case this code path will be executed? Running [this server](https://stackoverflow.com/a/40779258/2800559), I get `('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))`. This is not exactly `('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))` but pretty similar. When running `response = requests.get("http://localhost:5500/")`, I always get an exception which means that `response` is always undefined.

Based on that, it feels like `should_retry` will never get called in the case of a `Connection aborder`

Reply:
Yes, you're correct, since the actual error is caught by the `socket` and raised there as well. Therefore we should patch the CDK to handle that since the source could still crash if the error is hit in the middle of the `read` we need to handle this type of error on the CDK lvl with try__catch block for `check, discover, read`.

Reply:
Then the CDK raises the AirbyteTracebackException(type=transient) and this one is retryable for sure.

Reply:
I think this makes sense. This should be part of @pnilan's work on the HttpRequester. Should we put this on hold than and have you and @strosek work on the Shopify API migration?